### PR TITLE
[fftw3] Add fftw3 port to vcpkg

### PR DESCRIPTION
**Pull request by nagua**
Hello,

I made a fftw3 port for vcpkg.
It currently builds a SSE/SSE2 enabled version of fftw.
It can be build statically and dynamically.
I have not tested the static version in a project. But the library seems fine.

Greetings,
Nagua

### CODE REVIEW

File: ports/fftw3/config.h [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+#define SIZEOF_UNSIGNED_INT 4
+
+/* The size of `unsigned long', as computed by sizeof. */
+#define SIZEOF_UNSIGNED_LONG 4
+
+/* The size of `unsigned long long', as computed by sizeof. */
+#define SIZEOF_UNSIGNED_LONG_LONG 8
+
+/* The size of `void *', as computed by sizeof. */
+#define SIZEOF_VOID_P 4
```
Comment (codicodi) [inline ports/fftw3/config.h]: Looks like the `config.h` was generated for 32-bit build. Have you tried building for x64-* triplets? If library actually uses value of this macro things may go bad. In this case I'd recommend generating second `config.h` and either merge them based of whether `_WIN64` is defined or provide both and copy the correct one in the portfile.

Comment (nagua) [inline ports/fftw3/config.h]: That's interesting. I think there is indeed a problem. I checked the source and there is a usage in kernel/ifftw.h . So there might be a problem. But the config file is from an linked package from their website. So I have to look how to generate the file for x64.

Comment (codicodi) [inline ports/fftw3/config.h]: That good, but there are several other macros that could be affected (SIZEOF_SIZE_T, SIZEOF_PTRDIFF_T, maybe something else). If you are confident those are also not used, then I guess we could ignore them. 
BTW `HAVE_SNPRINTF` should be set - it is implemented in VS2015 and above.

Comment (nagua) [inline ports/fftw3/config.h]: Ok, now I tried to undef everything with sizes and averything compiles fine. So I think we can ignore the values for now.

**Comment by ras0219-msft** [comment]
Great job, thanks!

I made a modification while merging to avoid completely replacing the `CMAKE_CXX_FLAGS` variable. Instead, I just added `/bigobj` and `/openmp` to the `add_definitions()` directive.

**Pull request merged**